File: server/src/main/com/broll/gainea/server/core/cards/impl/direct/C_KillRandomSoldier.java
package com.broll.gainea.server.core.cards.impl.direct;

import com.broll.gainea.server.core.cards.DirectlyPlayedCard;
import com.broll.gainea.server.core.utils.PlayerUtils;
import com.broll.gainea.server.core.utils.StreamUtils;
import com.broll.gainea.server.core.utils.UnitControl;

import java.util.concurrent.atomic.AtomicInteger;

public class C_KillRandomSoldier extends DirectlyPlayedCard {

    private final static int COUNT = 3;

    public C_KillRandomSoldier() {
        super(42, "Gemetzel", "Verursacht 1 Schaden and jeder " + COUNT + "ten Einheit jedes Spielers.");
        setDrawChance(0.4f);
    }

    @Override
    protected void play() {
        PlayerUtils.iteratePlayers(game, 500, player -> {
            AtomicInteger integer = new AtomicInteger(0);
            StreamUtils.safeForEach(player.getUnits().stream().filter(it -> integer.incrementAndGet() % COUNT == 0),
                    unit -> UnitControl.damage(game, unit, 1));
        });
    }

}

File: server/src/main/com/broll/gainea/server/core/cards/impl/direct/C_KillRandomSoldier.kt
package com.broll.gainea.server.core.cards.impl.direct

import com.broll.gainea.server.core.cards.DirectlyPlayedCard
import com.broll.gainea.server.core.cards.EffectType.CHAOS
import com.broll.gainea.server.core.utils.UnitControl.damage
import com.broll.gainea.server.core.utils.iteratePlayers
import java.util.concurrent.atomic.AtomicInteger

class C_KillRandomSoldier : DirectlyPlayedCard(
    42,
    CHAOS,
    "Gemetzel",
    "Verursacht 1 Schaden and jeder " + COUNT + "ten Einheit jedes Spielers."
) {
    init {
        drawChance = 0.4f
    }

    override fun play() {
        with(game) {
            iteratePlayers(500) { player ->
                val integer = AtomicInteger(0)
                player.units.toList().filter { integer.incrementAndGet() % COUNT == 0 }
                    .forEach { damage(it, 1) }
            }
        }
    }

    companion object {
        private const val COUNT = 3
    }
}
